### ISSUE TITLE
When should I use the Instructor.Validator?

### ISSUE THREAD

**Issue by dfalling** [thmsmlr/instructor_ex#52]
Hi, first off thank you so much- this has greatly simplified and improved the success rate of parsing I was already doing with ChatGPT.

I'm a little confused however about when to use the `use Instructor.Validator` import. I currently added it in every relevant Ecto schema and the parent module that makes the call to `chat_completion`. Should it be in all of these places? Thanks in advance!

Edit: one note of clarification- I'm using a nested changeset/schema structure right now and it's working great. I have three levels of nested schemas with `changeset/2` functions. I haven't implemented the `valdidate_changeset` function in any of them, but have still seen the library prompt ChatGPT when it returned an improperly formatted field of the third-tier schema. Do I even need the `Instructor.Validator` and `validate_changeset`?

**Comment by acalejos** [comment]
My understanding is that if all you need is for the LLM's answer to be coerced into the schema's data type that you define then you don't need the `use Instructor.Validator`. It's mainly if you want to perform extra validations in addition to the type validations that you get with every call to `Instructor.chat_completion`. 

TLDR: It's only to register the `validate_changeset` callback so if you don't use that then you don't need to `use Instructor.Validator`

**Comment by dfalling** [comment]
Thanks for the response! Ah yeah, debugging my `changeset` function I see that it's actually not used by Instructor. I'm not sure how Instructor was so successful to reprompt the LLM when data was invalid without even using my changeset, but it was.

I've added the `use Instructor.Validator` to my schema, and a `validate_changeset` function that just passes through to my standard `changeset` function and I can see it's now being called.

Thanks!
